yarn-project/circuits.js/src/hash/map_slot.ts

```diff
@@ -0,0 +1,18 @@
+import { pedersenHash } from '@aztec/foundation/crypto';
+import { type Fr } from '@aztec/foundation/fields';
+
+/**
+ * Computes the resulting storage slot for an entry in a map.
+ * @param mapSlot - The slot of the map within state.
+ * @param key - The key of the map.
+ * @returns The slot in the contract storage where the value is stored.
+ */
+export function deriveStorageSlotInMap(
+  mapSlot: Fr | bigint,
+  key: {
+    /** Serialize to a field. */
```

Comment:
```suggestion
    /** Convert key to a field. */
```
Serialization would potentially be to many fields, if the value was larger. This is for things like address -> field.